baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreferImmutableStreamExCollections.java

```diff
@@ -0,0 +1,90 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.Collections;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "PreferImmutableStreamExCollections",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = SeverityLevel.WARNING,
+        summary = "Prefer immutable/unmodifable collections wherever possible because they are innately threadsafe, "
+                + "and easier to reason about when passed between different functions."
+                + " If you really want a mutable output then explicitly suppress this check.")
+public final class PreferImmutableStreamExCollections extends BugChecker
+        implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Matcher<ExpressionTree> TO_MAP = MethodMatchers.instanceMethod()
+            .onExactClass("one.util.streamex.EntryStream")
+            .named("toMap")
+            .withParameters(Collections.emptyList());
+
+    private static final Matcher<ExpressionTree> TO_SET = MethodMatchers.instanceMethod()
+            .onDescendantOf("one.util.streamex.AbstractStreamEx")
+            .named("toSet")
+            .withParameters(Collections.emptyList());
+
+    private static final Matcher<ExpressionTree> TO_LIST = MethodMatchers.instanceMethod()
+            .onDescendantOf("one.util.streamex.AbstractStreamEx")
+            .named("toList")
+            .withParameters(Collections.emptyList());
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        if (TO_MAP.matches(tree, state)) {
+            String base = state.getSourceForNode(ASTHelpers.getReceiver(tree.getMethodSelect()));
+            return buildDescription(tree)
+                    .setMessage("Prefer .toImmutableMap()")
```

Comment:
Nit: this is the only message most people see, I'd exclude overriding the more verbose message from the class definition since that explains the intent.